Basic/DyanmicPrefabs/Assets/Scripts/01_Server authoritative dynamic prefab loading across peers/DynamicPrefabManager.cs

```diff
@@ -0,0 +1,476 @@
+            });
+            var transport = m_NetworkManager.GetComponent<UnityTransport>();
+            transport.SetConnectionData(connectAddress, port);
+            
+            var payloadBytes = System.Text.Encoding.UTF8.GetBytes(payload);
+
+            m_NetworkManager.NetworkConfig.ConnectionData = payloadBytes;
+            m_NetworkManager.StartClient();
+            m_NetworkManager.CustomMessagingManager.RegisterNamedMessageHandler(nameof(ReceiveServerToClientSetDisconnectReason_CustomMessage), ReceiveServerToClientSetDisconnectReason_CustomMessage);
+        }
+
+        public void StartHost(string connectAddress, ushort port)
+        {
+            Debug.Log(nameof(StartHost));
+            m_ConnectAddress = connectAddress;
+            m_Port = port;
+            m_NetworkManager.NetworkConfig.ForceSamePrefabs = false;
+            var transport = m_NetworkManager.GetComponent<UnityTransport>();
+            transport.SetConnectionData(connectAddress, port);
+            m_NetworkManager.ConnectionApprovalCallback = ConnectionApprovalCallback;
+            m_NetworkManager.StartHost();
+        }
+
+        public override void OnDestroy()
+        {
+            UnloadAndReleaseAllDynamicPrefabs();
+            base.OnDestroy();
+        }
+
+        async void ReceiveServerToClientSetDisconnectReason_CustomMessage(ulong clientID, FastBufferReader reader)
+        {
+            reader.ReadValueSafe(out ConnectStatus status);
+
+            switch (status)
+            {
+                case ConnectStatus.Undefined:
+                    Debug.Log("Undefined");
+                    m_NetworkManager.Shutdown();
+                    break;
+                case ConnectStatus.ClientNeedsToPreload:
+                {
+                    m_NetworkManager.Shutdown();
+                    Debug.Log("Client needs to preload");
+                    reader.ReadValueSafe(out AddressableGUIDCollection addressableGUIDCollection);
+                    await LoadDynamicPrefabs(addressableGUIDCollection);
+                    Debug.Log("Restarting client");
+                    StartClient(m_ConnectAddress, m_Port);
+                }
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+
+            OnConnectionStatusReceived?.Invoke(status, reader);
+        }
+        
+        /// <summary>
+        /// Sends a DisconnectReason to the indicated client. This should only be done on the server, prior to disconnecting the client.
+        /// </summary>
+        /// <param name="clientID"> id of the client to send to </param>
+        /// <param name="status"> The reason for the upcoming disconnect.</param>
+        /// <param name="addressableGUIDCollection"></param>
+        void SendServerToClientSetDisconnectReason(ulong clientID, ConnectStatus status, AddressableGUIDCollection addressableGUIDCollection)
```

Comment:
@fernando-cortez With NGO 1.2 this should be done in the connection approval directly, not with a custom message